providers/base/units/stress/s3s4.pxu

```diff
@@ -150,3 +150,221 @@ requires: cpuinfo.platform in ("i386", "x86_64")
 command: [ -e "$PLAINBOX_SESSION_SHARE"/hibernate_{s4_iterations}_cycles.log ] && cat "$PLAINBOX_SESSION_SHARE"/hibernate_{s4_iterations}_cycles.log
 _description:
  Attaches the log from the {s4_iterations} cycles Hibernate/Resume test if it exists
+
+id: stress_s3_cycles_iterations_1
+category_id: stress-tests/suspend
+summary:
+  Resource job meant to be used in the S3 cycles stress templates
+plugin: resource
+environ: STRESS_S3_ITERATIONS STRESS_SUSPEND_REBOOT_ITERATIONS
+command:
+ REBOOT_ITERATIONS=${STRESS_SUSPEND_REBOOT_ITERATIONS:-3}
+ for i in $(seq 2 "$REBOOT_ITERATIONS"); do
+     echo "suspend_reboot_id: $i"
+     echo "suspend_reboot_previous: $((i-1))"
+     python3 -c 'import platform;print("fwts: {}".format("supported" if platform.machine() in ["x86_64", "i386"] else "unsupported"))'
+     echo
+ done
+estimated_duration: 1s
+flags: preserve-locale
+
+id: stress_s3_cycles_iterations_multiple
+category_id: stress-tests/suspend
+summary:
+  Resource job meant to be used in the S3 cycles stress templates
+plugin: resource
+environ: STRESS_S3_ITERATIONS STRESS_SUSPEND_REBOOT_ITERATIONS
+command:
+ REBOOT_ITERATIONS=${STRESS_SUSPEND_REBOOT_ITERATIONS:-3}
+ SUSPEND_ITERATIONS=${STRESS_S3_ITERATIONS:-30}
+ for i in $(seq 1 "$REBOOT_ITERATIONS"); do
+   for j in $(seq 2 "$SUSPEND_ITERATIONS"); do
+     echo "suspend_id: $j"
+     echo "suspend_id_previous: $((j-1))"
+     echo "suspend_reboot_id: $i"
+     python3 -c 'import platform;print("fwts: {}".format("supported" if platform.machine() in ["x86_64", "i386"] else "unsupported"))'
+     echo
+   done
+ done
+estimated_duration: 1s
+flags: preserve-locale
+
+id: stress_suspend_reboot_cycles_iterations
+category_id: stress-tests/suspend
+summary:
+  Resource job meant to be used in the suspend stress cycles of reboot template
+plugin: resource
+environ: STRESS_S3_ITERATIONS STRESS_SUSPEND_REBOOT_ITERATIONS
+command:
+ REBOOT_ITERATIONS=${STRESS_SUSPEND_REBOOT_ITERATIONS:-3}
+ SUSPEND_ITERATIONS=${STRESS_S3_ITERATIONS:-30}
+ for i in $(seq 1 "$REBOOT_ITERATIONS"); do
+   echo "s3_iterations: ${SUSPEND_ITERATIONS}"
+   echo "suspend_reboot_id: $i"
+   echo
+ done
+estimated_duration: 1s
+flags: preserve-locale
+
+id: stress_suspend_reboot_iterations
+category_id: stress-tests/suspend
+summary:
+  Resource job meant to be used in the suspend stress cycles of logcheck timecheck templates
+plugin: resource
+environ: STRESS_S3_ITERATIONS STRESS_SUSPEND_REBOOT_ITERATIONS STRESS_SUSPEND_SLEEP_THRESHOLD STRESS_SUSPEND_RESUME_THRESHOLD
+command:
+ REBOOT_ITERATIONS=${STRESS_SUSPEND_REBOOT_ITERATIONS:-3}
```

Comment:
I don't see the value in creating those intermediate variables? What am I missing?

Reply:
It will be used at below: L220: `echo "reboot_iterations: ${REBOOT_ITERATIONS}"`
And this `reboot_iterations` will be used in `id: stress-tests/suspend-{s3_iterations}-cycles-with-reboot-{reboot_iterations}-log-check` and `id: stress-tests/suspend-{s3_iterations}-cycles-with-reboot-{reboot_iterations}-log-attach`

Reply:
I don't think this answers my question.

From what I can tell, the code:
```
REBOOT_ITERATIONS=${STRESS_SUSPEND_REBOOT_ITERATIONS:-3}
 SUSPEND_ITERATIONS=${STRESS_S3_ITERATIONS:-30}
 SLEEP_THRESHOLD=${STRESS_SUSPEND_SLEEP_THRESHOLD:-10}
 RESUME_THRESHOLD=${STRESS_SUSPEND_RESUME_THRESHOLD:-5}
 echo "reboot_iterations: ${REBOOT_ITERATIONS}"
 echo "s3_iterations: ${SUSPEND_ITERATIONS}"
 echo "resume_threshold: ${RESUME_THRESHOLD}"
 echo "sleep_threshold: ${SLEEP_THRESHOLD}"
```
is equivalent to the:
```
 echo "reboot_iterations: ${STRESS_SUSPEND_REBOOT_ITERATIONS:-3}"
 echo "s3_iterations: ${STRESS_S3_ITERATIONS:-30}"
 echo "resume_threshold: ${STRESS_SUSPEND_SLEEP_THRESHOLD:-10}"
 echo "sleep_threshold: ${STRESS_SUSPEND_RESUME_THRESHOLD:-5}"
```

Which is 2x shorter and doesn't introduce extra variables.